starcli/layouts.py

```diff
@@ -34,75 +37,59 @@ def shorten_count(number):
+        yield title_table
+        yield ""
+        # Language
+        language = repo["language"]
+        if language:
+            yield Text(language, style="bold cyan")
+        else:
+            yield "[i cyan]unknown language"
+        yield ""
+        # Descripion
+        description = repo["description"]
+        if description:
+            yield Text(description.strip(), style="green")
+        else:
+            yield "[i green]no description"
+        yield ""
 
-    console.print(separator, justify="center", end="\n\n")  # print the separation
+    def column(renderable):
+        """Constrain width and align to center to create a column."""
+        return Align.center(renderable, width=LAYOUT_WIDTH, pad=False)
 
+    console = Console()  # initialise rich
     for repo in repos:
-
-        console.print(  # print the repo full name (user/repo)
-            " " * side_width,
-            f"[link={repo['html_url']}]{repo['full_name']}[/link]",
-            style="yellow",
-            end="  ",
-        )
-
-        stats = (  # construct the stats info
-            str(repo["stargazers_count"])
-            + "⭐, "
-            + str(repo["forks_count"])
-            + "🍴, "
-            + str(repo["watchers_count"])
-            + "👀"
-        )
-
-        if len(repo["full_name"] + stats) > len(separator + " "):
-            # if length of stats is not enough to print after the repo full name
-            print()  # print new line
-            console.print(  # and print the stats
-                " " * ((side_width) + (len(separator) - len(stats))),
-                stats,
-                end="\n\n",
-                style="blue",
-            )
-        else:
-            # if instead it is enough to print after repo full name
-            console.print(stats, end="\n\n", style="blue")  # then we print it
-
-        # print language
-        console.print(" " * side_width, repo["language"], style="bold cyan", end="\n\n")
-
-        # print description
-        console.print(
-            " " * side_width,
-            ("\n" + " " * side_width).join(
-                textwrap.wrap(f"{repo['description']}", len(separator))
-            ),
-            style="green",
-            end="\n\n",
-        )
-
-        # print separation
-        console.print(separator, justify="center", end="\n\n")
+        console.print(column(render_repo(repo)))
+    console.print(column(Rule()))
 
 
 def table_layout(repos):
     """ Displays repositories in a table format using rich """
 
-    table = Table()
+    table = Table(leading=1)
 
     # make the columns
-    table.add_column("Name", style="bold cyan", no_wrap=True, width=45)
-    table.add_column("Language", style="green", no_wrap=True, width=23)
-    table.add_column("Description", style="blue", no_wrap=True, width=140)
-    table.add_column("Stats", style="magenta", no_wrap=False, width=57)
+    table.add_column("Name", style="bold cyan")
```

Comment:
Table handles compressed columns better now. It will shrink the longest column first.